walletlibrarydemo/src/main/java/com/microsoft/walletlibrarydemo/feature/issuance/viewlogic/RequirementsFragment.kt

```diff
@@ -0,0 +1,193 @@
+            if (viewModel.verifiedIdResult?.getOrNull() != null) {
+                binding.requestTitle.text = "Presentation Complete!!"
+                binding.requestCompletion.visibility = View.GONE
+            }
+        } else
+            showError("Presentation Failed: ${viewModel.verifiedIdResult?.exceptionOrNull()}")
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    private fun configureVerifiedIdView() {
+        var verifiedId: VerifiedId? = null
+        viewModel.verifiedIdResult?.let {
+            if (it.isSuccess)
+                verifiedId = it.getOrNull() as VerifiedId
+        }
+        verifiedId?.let {
+            val verifiableCredential = verifiedId as VerifiableCredential
+            binding.requestTitle.text = verifiableCredential.types.last()
+            val claims = verifiableCredential.getClaims()
+            claims.add(VerifiedIdClaim("Issued On", verifiableCredential.issuedOn))
+            verifiableCredential.expiresOn?.let { claims.add(VerifiedIdClaim("Expiry", it)) }
+            claims.add(VerifiedIdClaim("Id", verifiableCredential.id))
+            val adapter = VerifiedIdAdapter(claims)
+            binding.verifiedIdClaims.layoutManager = LinearLayoutManager(context)
+            binding.verifiedIdClaims.isNestedScrollingEnabled = false
+            binding.verifiedIdClaims.adapter = adapter
+            binding.requestCompletion.visibility = View.GONE
+        }
+    }
+
+    private fun goBackHome() {
+        findNavController().navigate(RequirementsFragmentDirections.actionRequirementsFragmentToLoadRequestFragment())
+    }
+
+    override fun navigateToVerifiedIds(requirement: VerifiedIdRequirement) {
+        runBlocking {
+            binding.requirementsList.visibility = View.GONE
+            binding.verifiedIdClaims.visibility = View.GONE
+            binding.verifiedIds.visibility = View.VISIBLE
+            binding.matchingIds.visibility = View.VISIBLE
+            val decodedVerifiedIds = filterVerifiedIdsByType(requirement)
+            if (decodedVerifiedIds.isNotEmpty()) {
+                binding.matchingIds.text = "Matching Verified Ids:"
+                val adapter = VerifiedIdsAdapter(
+                    this@RequirementsFragment,
+                    decodedVerifiedIds,
+                    requirement
+                )
+                binding.verifiedIds.layoutManager = LinearLayoutManager(context)
+                binding.verifiedIds.isNestedScrollingEnabled = false
+                binding.verifiedIds.adapter = adapter
+            }
+        }
+    }
+
+    private suspend fun filterVerifiedIdsByType(requirement: VerifiedIdRequirement): ArrayList<VerifiedId> {
+        val decodedVerifiedIds = ArrayList<VerifiedId>()
+        viewModel.getVerifiedIds().forEach {
+            if ((it is VerifiableCredential) && it.types.contains(requirement.types.last())) {
```

Comment:
can't we use doesMatch for this instead?

Reply:
done.